code/datums/medal_awards.dm

```diff
@@ -366,3 +463,185 @@ GLOBAL_LIST_INIT(xeno_medals, list(XENO_SLAUGHTER_MEDAL, XENO_RESILIENCE_MEDAL,
+	recommendation.recommended_by_name = recommendation_giver.real_name
+	recommendation.recommended_by_ckey = recommendation_giver.ckey
+	recommendation.recommended_by_rank = recipient_ranks[recommendation_giver.real_name]
+
+
+	recommendation.reason = reason
+
+	return TRUE
+
+
+GLOBAL_DATUM_INIT(ic_medals_panel, /datum/ic_medal_panel, new)
+
+/datum/ic_medal_panel
+	var/name = "Medals Panel"
+	var/list/user_locs = list()
+
+/datum/ic_medal_panel/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "IcMedalsPanel", "Medals Panel")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/ic_medal_panel/ui_state(mob/user)
+	return GLOB.not_incapacitated_state
+
+/datum/ic_medal_panel/ui_data(mob/user)
+	var/list/data = list()
+	data["recommendations"] = list()
+
+	for(var/datum/medal_recommendation/recommendation in GLOB.medal_recommendations)
+		var/recommendation_list = list()
+
+		recommendation_list["rank"] = recommendation.recipient_rank
+		recommendation_list["name"] = recommendation.recipient_name
+		recommendation_list["ref"] = REF(recommendation)
+		recommendation_list["recommender_name"] = recommendation.recommended_by_name
+		recommendation_list["reason"] = recommendation.reason
+		recommendation_list["recommender_rank"] = recommendation.recommended_by_rank
+
+		data["recommendations"] += list(recommendation_list)
+	return data
+
+/datum/ic_medal_panel/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/carbon/human/user = usr
+	var/obj/item/card/id/card = user.wear_id
+	if(!card)
+		to_chat(user, SPAN_WARNING("You must have an authenticated ID Card to award medals."))
+		return
+
+	if(!((card.paygrade in GLOB.co_paygrades) || (card.paygrade in GLOB.highcom_paygrades)))
+		to_chat(user, SPAN_WARNING("Only a Senior Officer can award medals!"))
+		return
+
+	if(!card.registered_ref)
+		user.visible_message("ERROR: ID card not registered in USCM registry. Potential medal fraud detected.")
+		return
+
+	var/real_owner_ref = card.registered_ref
+
+	if(real_owner_ref != WEAKREF(user))
+		user.visible_message("ERROR: ID card not registered for [user.real_name] in USCM registry. Potential medal fraud detected.")
+		return
+
+	switch(action)
+		if("grant_new_medal")
+			if(!user_locs[user])
+				return
+			if(give_medal_award(get_turf(user_locs[user])))
+				var/atom/user_machine = user_locs[user]
+				user_machine.visible_message(SPAN_NOTICE("[user_machine] prints a medal."))
+			. = TRUE
+
+		if("approve_medal")
+			var/recommendation_ref = params["ref"]
+			var/medal_type = params["medal_type"]
+			if(!(medal_type in GLOB.human_medals))
+				return
+			var/datum/medal_recommendation/recommendation = locate(recommendation_ref) in GLOB.medal_recommendations
+			if(!recommendation)
+				return
+			if(!user_locs[user])
+				return
+			if(recommendation.recipient_name == user.real_name)
+				return
```

Comment:
They will probably figure out why this doesn't work at some point, but you should to_chat a message why this doesn't work. As is it just makes the UI seem like its bugged because it doesn't do anything when you press the button.

Reply:
This is a sanity check. It shouldn't be a

Reply:
It should remove their whitelist instead